File: src/main/java/com/gmail/borlandlp/minigamesdtools/gui/hotbar/items/inventory_gui/InventoryPageOpenRequesterCreator.java
package com.gmail.borlandlp.minigamesdtools.gui.hotbar.items.inventory_gui;

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import com.gmail.borlandlp.minigamesdtools.config.ConfigEntity;
import com.gmail.borlandlp.minigamesdtools.config.ConfigPath;
import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider;
import com.gmail.borlandlp.minigamesdtools.creator.Creator;
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo;
import org.bukkit.Material;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.inventory.ItemStack;

import java.util.ArrayList;
import java.util.List;

@CreatorInfo(creatorId = "inventory_page_open_requester")
public class InventoryPageOpenRequesterCreator implements Creator {
    @Override
    public List<String> getDataProviderRequiredFields() {
        return new ArrayList<>();
    }

    @Override
    public InventoryPageOpenRequester create(String ID, AbstractDataProvider dataProvider) throws Exception {
        InventoryPageOpenRequester requester = new InventoryPageOpenRequester();
        ConfigEntity configEntity = MinigamesDTools.getInstance().getConfigProvider().getEntity(ConfigPath.HOTBAR_SLOTS, ID);
        ConfigurationSection configurationSection = configEntity.getData();

        if(dataProvider.contains("page_id")) {
            requester.setPageId((String) dataProvider.get("page_id"));
        } else {
            requester.setPageId(configEntity.getData().get("page_id").toString());
        }

        ItemStack activeIcon = new ItemStack(Material.getMaterial(configurationSection.get("active_icon").toString()));
        requester.setActiveIcon(activeIcon);
        ItemStack unactiveIcon = new ItemStack(Material.getMaterial(configurationSection.get("unactive_icon").toString()));
        requester.setUnactiveIcon(unactiveIcon);
        requester.setCooldownTime(Integer.parseInt(configurationSection.get("cooldown").toString()));
        requester.setInfiniteSlot(Boolean.parseBoolean(configurationSection.get("infinite").toString()));
        requester.setAmount(Integer.parseInt(configurationSection.get("amount").toString()));

        return requester;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/gui/hotbar/items/inventory_gui/InventoryPageOpenRequesterCreator.kt
package com.gmail.borlandlp.minigamesdtools.gui.hotbar.items.inventory_gui

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools
import com.gmail.borlandlp.minigamesdtools.config.ConfigPath
import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider
import com.gmail.borlandlp.minigamesdtools.creator.Creator
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo
import org.bukkit.Material
import org.bukkit.inventory.ItemStack
import java.util.*

@CreatorInfo(creatorId = "inventory_page_open_requester")
class InventoryPageOpenRequesterCreator : Creator() {
    @Throws(Exception::class)
    override fun create(id: String, dataProvider: AbstractDataProvider): InventoryPageOpenRequester {
        val requester = InventoryPageOpenRequester()
        val configEntity =
            MinigamesDTools.instance!!.configProvider!!.getEntity(ConfigPath.HOTBAR_SLOTS, id)
        val configurationSection = configEntity!!.data
        if (dataProvider.contains("page_id")) {
            requester.pageId = dataProvider["page_id"] as String
        } else {
            requester.pageId = configEntity.data["page_id"].toString()
        }
        val activeIcon =
            ItemStack(Material.getMaterial(configurationSection["active_icon"].toString()))
        requester.activeIcon = activeIcon
        val unactiveIcon =
            ItemStack(Material.getMaterial(configurationSection["unactive_icon"].toString()))
        requester.unactiveIcon = unactiveIcon
        requester.setCooldownTime(configurationSection["cooldown"].toString().toInt().toLong())
        requester.isInfiniteSlot = configurationSection["infinite"].toString().toBoolean()
        requester.amount = configurationSection["amount"].toString().toInt()
        return requester
    }
}